### DISCUSSION TITLE
Current status of Iran

### DISCUSSION THREAD

**Discussion by MortThePersian** [XTLS/Xray-core#1961] [General]
Hello there;

I felt an explanation needed to show a more clear picture of Iran restrictions situation.

I would realy like to describe or present this with scientific approach but unfortunately this option is not available, at least not yet !

Let Me describe the current status of restrictions in a compacted frame;

- Almost no standard VPN protocol will work in here now despite the method you use.
Only two way in this procedure will have limited success, 
OpenVpn / TCP + stunnel with obfuscated & Cipher 256
Using stunnel with Iranian providers to tunnel yourself out that have high risk to you & all people using the service & it is a proven fact !

- Beside v2ray, Geph that based on sosistab protocol have shown signs of success but it's not yet competent enough option

- In v2ray itself, 
Shadowsocks will be blocked by approximately 24 hours 
VMess may face problem with time & becomes obsolete
Vless & Trojan is the best choice now & Vless have shown to be even better

- For transmission protocol, We have no choice but to use CDN & of course Cloudflare is the best option.
Due to that reason, WS & gRPC are the way to go...
From experiences, WS is more stable & gRPC is faster overall. mode option in gRPC, "gun" mode resulted in better stability in tests.

- Vless Vision Reality is what We tested & Could bypass the restrictions successfully with a satisfying result

- Currently, Mobile providers have tangibly more tighter restrictions

- Different providers have different restrictions

- Restrictions may vary in the different regions of the Nation 

Overall, Reality is the only solution that most have agreed on & many are desperately waiting for the release.

◇ TLS is a must, best result from experiences is, min 1.0 1.2 in the panel & 1.3 only in the Cloudflare settings.

◇ For the Cloudflare, We also Cname the Subdomain with proxy on because many VPS IP We get are blocked from the start !
& also in general SSL settings We choose, Full or Strict (other options will not work or will result in unstable connection)

◇ Redirecting your proxied Subdomain to a well-known website have shown to be effective for delaying the blockade 

◇ Using fully single port 443 have shown better results overall

~ Some users reported that they had achieved better results with a paid SSL Certificate

~ Some users reported that different Nameservers have shown to be a factor, Specially in the last days !

In the past days, one of the daily tasks of life is finding the working Cloudflare IP !
& if you find one that works, You probably still have an issue with uploading speed which is near 0 ...

**Comment by computerscot** [comment]
What is the software in "OpenVpn TCP stunnel with obfuscated & Cipher 256"? 

Do you mean OpenVPN + stunnel + obfs4proxy?

**Reply by MortThePersian** [reply]
Yes, I will edit & add the "+" to avoid the misunderstanding could occur among the others as well.
Thank you !

**Comment by RPRX** [comment]
情况确实比较严重，~~虽然仔细想想我们的情况也没好多少~~，Xray-core v1.8.1 已预发布，需要公测一段时间，没问题的话就会变为正式版

REALITY 的 Web 面板、一键脚本、各种教程、客户端等周边支持已经非常丰富了，~~所以不必 desperately waiting for the release~~

Xray-core 的 README/Usage/Tutorial 已添加 https://github.com/SasukeFreestyle/XTLS-Iran-Reality ，~~其实 Web 面板和一键脚本更方便~~

**Reply by MortThePersian** [reply]
In fact, You may face the complications We experiencing now !
The problem We have in here is that old folks don't know much about tech stuff & are struggling with every piece of new thing so We have to go for the easiest & most stable options
For example, We make subscription links so they just have to go into the app & update the config
Because of Reality is still in beta, that progress becomes complex, at least a little bit ...
& I really don't want to see the old man with teary eyes again that turning on & off the v2ray app in hope to see his grandchildren abroad 
The desperate one is actually Me !

P.S. I'm not a seller

**Reply by yumeo-ch** [reply]
Hey friend, I know this is inappriopriate, but I'm deseprately asking you to help me personally for setting up v2ray reality, 
im not a complete noob but not an expert either, something in the middle.
where can I reach you if you accept ? I couldn't sleep since 3 days in order to set up these proxy, have some pity :D

**Reply by MortThePersian** [reply]
Hello there,
Don't worry at all ! from the developers to communities, all are here to help...
here is a quick list;

official Xray-core channel

https://t.me/projectXray

official 3x-ui panel channel 

https://t.me/panel3xuiGroup

official Gozargah / Marzban panel channel 

https://t.me/gozargah_marzban

& of course 

https://www.certifiedtruecopy.de/

https://t.me/+9UrelMPPBiIxY2Zi

* except the Xray-core official, all others are Persian

**Comment by RPRX** [comment]
我比较好奇伊朗的最终目标，它是想实现完全的局域网，还是仍允许一些境外的大公司向境内的个人直接提供服务？哪些类型的服务？

**Reply by hawshemi** [reply]
Something between China and North Korea.
That's the goal.

**Reply by computerscot** [reply]
There is a publication called _Iran International_, based in Washington, DC, that sometimes reports on the Internet situation in Iran.

They published an article on 2022-12-18 titled "Frustrated Over Controlling Internet, Iran May Cut It Altogether." The regime has two goals: (1) stopping the circulation of news; and (2) stopping Iranians from communicating with each other. All that prevents the regime from completely cutting off the Internet from the rest of the world is the needs of the banking sector, says the article.

On 2023-03-18, _Iran International_ published another article, "Iranians Sustain Huge Losses Due To Government Internet Shutdowns." It says that people who work online are losing their jobs. Tech companies are losing sales. 

Apparently this does not matter to the regime. Only the big financial companies have any influence with the government.

**Reply by MortThePersian** [reply]
In addition to what others said, They ultimate goal is would be shut down the Internet altogether, 
Although it may seem unbelievable just remember that in here you may die because of not having a proper hijab ! & schools are under attack with the chemical weapons...
Of course they are hypocritic, Internet will always be available to themselves throughout some sort of exclusive infrastructure. As they was still online when they shut down the entire Internet once on the peak of the never-ending protests.
Some of them even mocked others & their mercenaries in news agencies tried to cover up the incident !
Now, the situation is not far from the total shut down either at all !

**Comment by yumeo-ch** [comment]
> Cname the Subdomain

By subdomain, do you mean vpn.example.com ?
do we have to have a subdomain ? can't we directly the domain as it is for running reality on it ?
I mean, I didn't get why there must be a subdomain.

**Reply by MortThePersian** [reply]
There are other methods as well, just have a quick look at the groups I mentioned earlier in a reply to your other post
Try to test them, Currently there is no guaranteed way to point out...

**Comment by yumeo-ch** [comment]
> ~ Some users reported that they had achieved better results with a paid SSL Certificate

Why do we need a paid ssl certificate with reality,
isn't reality supposed to steal anohter domain's certificate ?

**Reply by MortThePersian** [reply]
It's not for Reality !
& this is a quote based on user tests & experiences 
I just tried to make a more clear picture of restrictions to the people who are outside of Iran & of course developers, So they could help Us in a more efficient way if there is a possibility...
Currently, Just test every option you have & try to actively share your achievements & update your knowledge through Iranian communities.

**Reply by yumeo-ch** [reply]
yes, ideally we need somehting that works on every operator, a proper opensource method that can't be blocked by a robot.
